crates/network/src/ethereum/builder.rs

```diff
@@ -127,30 +127,34 @@ where
     N::UnsignedTx: From<TxLegacy> + From<TxEip1559> + From<TxEip2930> + From<TxEip4844Variant>,
 {
     match (
-        request.gas_price.as_ref(),
-        request.max_fee_per_gas.as_ref(),
         request.access_list.as_ref(),
         request.max_fee_per_blob_gas.as_ref(),
         request.blob_versioned_hashes.as_ref(),
         request.sidecar.as_ref(),
     ) {
-        // Legacy transaction
-        (Some(_), None, None, None, None, None) => build_legacy(request).map(Into::into),
-        // EIP-2930
-        // If only accesslist is set, and there are no EIP-1559 fees
-        (_, None, Some(_), None, None, None) => build_2930(request).map(Into::into),
-        // EIP-1559
-        // If EIP-4844 fields are missing
-        (None, _, _, None, None, None) => build_1559(request).map(Into::into),
-        // EIP-4844
-        // All blob fields required
-        (None, _, _, Some(_), Some(_), Some(_)) => {
-            build_4844(request).map(TxEip4844Variant::from).map(Into::into)
+        (_, Some(_), None, _) => Err(TransactionBuilderError::MissingKey("blob_versioned_hashes")),
```

Comment:
can we turn these into functions as well? even with 2 things gone, this match block is really bad code smell